src/Editor.js

```diff
@@ -0,0 +1,127 @@
+    this.editPanel = null;
+
+    this.pointerUpHandlerPlaneFinder = this.onPointerUpPlaneFinder.bind(this);
+
+    this.initialized = false;
+    this.init();
+  }
+
+  init() {
+    if (this.initialized) return;
+
+    this.setupEditPanel();
+
+    this.initialized = true;
+  }
+
+  setupEditPanel() {
+    this.editPanel = document.createElement('div');
+    this.editPanel.style.position = 'absolute';
+    this.editPanel.style.padding = '10px';
+    this.editPanel.style.backgroundColor = '#cccccc';
+    this.editPanel.style.border = '#aaaaaa 1px solid';
+    this.editPanel.style.zIndex = 90;
+    this.editPanel.style.width = '375px';
+    this.editPanel.style.fontFamily = 'arial';
+    this.editPanel.style.fontSize = '10pt';
+    this.editPanel.style.textAlign = 'left';
+
+
+    const editTable = document.createElement('div');
+    editTable.style.width = '100%';
+
+
+    // Plane Finder
+    const planeFinder = document.createElement('div');
+    planeFinder.style.width = '100%';
+    planeFinder.style.display = 'flex';
+    planeFinder.style.flexDirection = 'row';
+    planeFinder.style.justifyContent = 'space-between';
+
+    const planeFinderLabel = document.createElement('p');
+    planeFinderLabel.id = 'planeFinderLabel';
+    planeFinderLabel.innerHTML =
+      `Up: ${this.viewer.camera.up.x.toFixed(3)}, ${this.viewer.camera.up.y.toFixed(3)}, ${this.viewer.camera.up.z.toFixed(3)}`;
+
+    const planeFinderButton = document.createElement('button');
+    planeFinderButton.innerHTML = 'Find ground plane';
+    planeFinderButton.addEventListener('click', () => {
+      this.setupPlaneFinder();
+    });
+
+    planeFinder.appendChild(planeFinderLabel);
+    planeFinder.appendChild(planeFinderButton);
+
+    editTable.appendChild(planeFinder);
+
+    this.editPanel.appendChild(editTable);
+    this.editPanel.style.display = 'block';
+    this.viewer.renderer.domElement.parentElement.prepend(this.editPanel);
+  }
+
+  setupPlaneFinder() {
+    if (this.viewer.useBuiltInControls) {
+      this.viewer.rootElement.removeEventListener('pointerup', this.viewer.pointerUpHandler);
+      this.viewer.rootElement.addEventListener('pointerup', this.pointerUpHandlerPlaneFinder);
+    }
+  }
+
+  teardownPlaneFinder() {
+    if (this.viewer.useBuiltInControls) {
+      this.viewer.rootElement.removeEventListener('pointerup', this.pointerUpHandlerPlaneFinder);
+      this.viewer.rootElement.addEventListener('pointerup', this.viewer.pointerUpHandler);
+    }
+  }
+
+  onPointerUpPlaneFinder = function() {
+    const renderDimensions = new THREE.Vector2();
+    const clickOffset = new THREE.Vector2();
+    // const toNewFocalPoint = new THREE.Vector3();
+    const outHits = [];
+    let points = [];
+
+    return function(mouse) {
+      clickOffset.copy(this.viewer.mousePosition).sub(this.viewer.mouseDownPosition);
+      const mouseUpTime = getCurrentTime();
+      const wasClick = mouseUpTime - this.viewer.mouseDownTime < 0.5 && clickOffset.length() < 2;
```

Comment:
I think we need to change `Viewer.onMouseUp()` to only do these calculations to determine if there was a click, and define a new function `Viewer.onClick()` and call that from `Viewer.onMouseUp()`. Then we could just override `Viewer.onClick()` and not have to replicate the click detection logic.

Reply:
Agreed